autogen/agentchat/conversable_agent.py

```diff
@@ -1114,10 +1115,17 @@ def my_summary_method(
         return summary
 
     @staticmethod
-    def _last_msg_as_summary(sender, recipient, summary_args) -> str:
+    def _last_msg_as_summary(sender, recipient, summary_args) -> Union[str, List]:
         """Get a chat summary from the last message of the recipient."""
         try:
-            summary = recipient.last_message(sender)["content"].replace("TERMINATE", "")
+            content = recipient.last_message(sender)["content"]
+            if isinstance(content, str):
+                summary = content.replace("TERMINATE", "")
+            elif isinstance(content, list):
+                summary = content
+                for component in summary:
+                    if "text" in component and isinstance(component, dict):
+                        component["text"] = component["text"].replace("TERMINATE", "")
```

Comment:
Tiny nit pick: Should we flip the order in this if statement? It might cause an unexpected error if something weird was passed in the list (currently we expect IndexError and AttributeError)
```suggestion
                    if isinstance(component, dict) and "text" in component:
                        component["text"] = component["text"].replace("TERMINATE", "")
```